src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -223,14 +221,22 @@ public static class PrereleaseCompatibilityTransformer
 
                     JObject tool = (JObject)run["tool"];
                     modifiedLog |= RenameProperty(tool, previousName: "fileVersion", newName: "dottedQuadFileVersion");
-                    if (tool != null && tool["language"] == null) { tool["language"] = "en-US"; }
+                    if (tool != null && tool["language"] == null)
+                    {
+                        tool["language"] = "en-US";
+                        modifiedLog = true;
```

Comment:
>modifiedLog [](start = 24, length = 11)

I noticed we missed a few of these. Maybe in future we can decorate the log with some sort of observer that notes any changes.

Reply:
or we could actually unit test the damn thing

---
In reply to: [248875783](https://github.com/Microsoft/sarif-sdk/pull/1221#discussion_r248875783) [](ancestors = 248875783)

Reply:
We could, but you can't test your way out of a fragile design ;-) The design requires us to remember to set this flag on every change. I sure don't want to get into this now, it's just something I noticed and spackled over while I was debugging through this code.

---
In reply to: [248881102](https://github.com/Microsoft/sarif-sdk/pull/1221#discussion_r248881102) [](ancestors = 248881102,248875783)

Reply:
whatever your thoughts on the design (agree there's a conversation here), it would be useful for us to be testing individual transformations that we add to this code with discrete test cases. that is a simple discipline that can avoid adding additional complexity. particularly for a potentially short-lived type (as we originally perceived this thing to be). 

---
In reply to: [248881460](https://github.com/Microsoft/sarif-sdk/pull/1221#discussion_r248881460) [](ancestors = 248881460,248881102,248875783)

Reply:
fairly simple change might achieve the value you have in mind, simply create a helper for every mutation operation, and set the variable there. not much of an actual design change required.

---
In reply to: [248883476](https://github.com/Microsoft/sarif-sdk/pull/1221#discussion_r248883476) [](ancestors = 248883476,248881460,248881102,248875783)

Reply:
I introduced `PopulatePropertyIfAbsent` which gets us part of the way there.

---
In reply to: [248884175](https://github.com/Microsoft/sarif-sdk/pull/1221#discussion_r248884175) [](ancestors = 248884175,248883476,248881460,248881102,248875783)